Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -423,6 +423,108 @@ std::optional<int64_t> parseTimeZoneNumericUTCOffset(StringView string)
     });
 }
 
+template<typename CharacterType>
+static std::optional<int64_t> parseUTCOffsetInMinutes(StringParsingBuffer<CharacterType>& buffer)
+{
+    // UTCOffset :::
+    //     TemporalSign Hour
+    //     TemporalSign Hour HourSubcomponents[+Extended]
+    //     TemporalSign Hour HourSubcomponents[~Extended]
+    //
+    // TemporalSign :::
+    //     ASCIISign
+    //     <MINUS>
+    //
+    // ASCIISign ::: one of
+    //     + -
+    //
+    // Hour :::
+    //     0 DecimalDigit
+    //     1 DecimalDigit
+    //     20
+    //     21
+    //     22
+    //     23
+    //
+    // HourSubcomponents[Extended] :::
+    //     TimeSeparator[?Extended] MinuteSecond
+    //
+    // TimeSeparator[Extended] :::
+    //     [+Extended] :
+    //     [~Extended] [empty]
+    //
+    // MinuteSecond :::
+    //     0 DecimalDigit
+    //     1 DecimalDigit
+    //     2 DecimalDigit
+    //     3 DecimalDigit
+    //     4 DecimalDigit
+    //     5 DecimalDigit
+
+    // sign and hour.
+    if (buffer.lengthRemaining() < 3)
+        return std::nullopt;
+
+    int64_t factor = 1;
+    if (*buffer == '+')
+        buffer.advance();
+    else if (*buffer == '-' || *buffer == minusSign) {
+        factor = -1;
+        buffer.advance();
+    } else
+        return std::nullopt;
+
+    unsigned hour = 0;
+    ASSERT(buffer.lengthRemaining() >= 2);
+    auto firstHourCharacter = *buffer;
+    if (firstHourCharacter >= '0' && firstHourCharacter <= '2') {
```

Comment:
Hmm, I think this would be nicer if you flipped the condition, so that you don't need `else return`.

Reply:
Changed.